src/screens/MoreScreen.js

```diff
@@ -1,33 +1,95 @@
+import {AppLoading} from 'expo';
 import React, {Component} from 'react';
-import {Text} from 'react-native';
-import {Icon} from 'react-native-elements';
+import {Platform, Text, View} from 'react-native';
+import {connect} from 'react-redux';
+import {Icon, Button, Card} from 'react-native-elements';
 
-import {CardSection, Card} from '../components/common/index';
+import {logUserOutOfFacebook} from '../actions';
+
+import {more_screen} from '../locale.en';
+import {navKeys} from '../constants';
 
 class MoreScreen extends Component {
+    componentDidMount() {
+        const {token, navigation: {navigate, state: {key}}} = this.props;
+
+        if (!token && key !== navKeys.LOGIN) {
+            navigate(navKeys.LOGIN);
+        }
```

Comment:
This pattern we'll probably want to implement on most screens. If there's no token, and we're not already on the `LOGIN` screen, then kick us back to `LOGIN`. (Note we have to check where we are since react-native renders all the screens at once, even if you're not on them.